core/src/test/scala/integration/kafka/api/AdminClientIntegrationTest.scala

```diff
@@ -759,15 +758,17 @@ class AdminClientIntegrationTest extends IntegrationTestHarness with Logging {
       // Need to retry if leader is not available for the partition
       result = client.deleteRecords(Map(topicPartition -> RecordsToDelete.beforeOffset(0L)).asJava)
 
+      lowWatermark = Long.MinValue
       val future = result.lowWatermarks().get(topicPartition)
       try {
-        lowWatermark = future.get(1000L, TimeUnit.MILLISECONDS).lowWatermark()
+        lowWatermark = future.get().lowWatermark()
         lowWatermark == 5L
       } catch {
-        case e: LeaderNotAvailableException => false
-      }
-
-    }, "Expected low watermark of the partition to be 5L")
+        case e: ExecutionException if e.getCause.isInstanceOf[LeaderNotAvailableException] ||
+          e.getCause.isInstanceOf[NotLeaderForPartitionException] => false
+        }
+    }, "Expected low watermark of the partition to be 5 but got ".concat(
```

Comment:
nit: a bit more idiomatic to use string interpolation. If we make `lowWatermark` an option, then we can simplify this to just 
```scala
s"Expected low watermark of the partition to be 5 but got $lowWatermark"
```